frontend/apps/filemanager/filemanagerfilesearcher.lua

```diff
@@ -45,7 +46,11 @@ function FileSearcher:readDir()
                 local fullpath = d.."/"..f
                 local attributes = lfs.attributes(fullpath) or {}
                 -- Don't traverse hidden folders if we're not showing them
-                if attributes.mode == "directory" and f ~= "." and f ~= ".." and (G_reader_settings:isTrue("show_hidden") or not util.stringStartsWith(f, ".")) then
+                if attributes.mode == "directory" and f ~= "." and f ~= ".."
+                    and (G_reader_settings:isTrue("show_hidden") or not util.stringStartsWith(f, "."))
+                    and FileChooser:show_dir(f)
+                    and FileChooser:show_file(f)
```

Comment:
Do you need to call the 2 show_dir AND show_file in places where you know it's already a dir or a file ?

Reply:
No,